tests/Unit/Evolution/DiscontinuousGalerkin/Actions/Test_ApplyBoundaryCorrections.cpp

```diff
@@ -0,0 +1,635 @@
+
+  // The reference element in 2d denoted by X below:
+  // ^ eta
+  // +-+-+> xi
+  // |X| |
+  // +-+-+
+  // | | |
+  // +-+-+
+  //
+  // The "self_id" for the element that we are considering is marked by an X in
+  // the diagram. We consider a configuration with one neighbor in the +xi
+  // direction (east_id), and (in 2d and 3d) one in the -eta (south_id)
+  // direction.
+  //
+  // In 1d there aren't any projections to test, and in 3d we only have 1
+  // element in the z-direction.
+  DirectionMap<Dim, Neighbors<Dim>> neighbors{};
+  ElementId<Dim> self_id{};
+  ElementId<Dim> east_id{};
+  ElementId<Dim> south_id{};  // not used in 1d
+
+  if constexpr (Dim == 1) {
+    self_id = ElementId<Dim>{0, {{{1, 0}}}};
+    east_id = ElementId<Dim>{0, {{{1, 1}}}};
+    neighbors[Direction<Dim>::upper_xi()] = Neighbors<Dim>{{east_id}, {}};
+  } else if constexpr (Dim == 2) {
+    self_id = ElementId<Dim>{0, {{{1, 0}, {0, 0}}}};
+    east_id = ElementId<Dim>{0, {{{1, 1}, {0, 0}}}};
+    south_id = ElementId<Dim>{1, {{{1, 0}, {0, 0}}}};
+    neighbors[Direction<Dim>::upper_xi()] = Neighbors<Dim>{{east_id}, {}};
+    neighbors[Direction<Dim>::lower_eta()] = Neighbors<Dim>{{south_id}, {}};
+  } else {
+    static_assert(Dim == 3, "Only implemented tests in 1, 2, and 3d");
+    self_id = ElementId<Dim>{0, {{{1, 0}, {0, 0}, {0, 0}}}};
+    east_id = ElementId<Dim>{0, {{{1, 1}, {0, 0}, {0, 0}}}};
+    south_id = ElementId<Dim>{1, {{{1, 0}, {0, 0}, {0, 0}}}};
+    neighbors[Direction<Dim>::upper_xi()] = Neighbors<Dim>{{east_id}, {}};
+    neighbors[Direction<Dim>::lower_eta()] = Neighbors<Dim>{{south_id}, {}};
+  }
+  const Element<Dim> element{self_id, neighbors};
+
+  MockRuntimeSystem runner{{std::vector<std::array<size_t, Dim>>{
+                                make_array<Dim>(2_st), make_array<Dim>(3_st)},
+                            std::make_unique<BoundaryTerms<Dim>>(),
+                            dg_formulation}};
+
+  const size_t number_of_grid_points_per_dimension = 5;
+  const Mesh<Dim> mesh{number_of_grid_points_per_dimension,
+                       Spectral::Basis::Legendre, quadrature};
+
+  ::InverseJacobian<DataVector, Dim, Frame::Logical, Frame::Inertial> inv_jac{
+      mesh.number_of_grid_points(), 0.0};
+  for (size_t i = 0; i < Dim; ++i) {
+    inv_jac.get(i, i) = 2.0;
+  }
+  const auto det_inv_jacobian = determinant(inv_jac);
+  const auto jacobian = determinant_and_inverse(inv_jac).second;
+
+  const auto logical_coords = logical_coordinates(mesh);
+  tnsr::I<DataVector, Dim, Frame::Inertial> inertial_coords{};
+  for (size_t i = 0; i < logical_coords.size(); ++i) {
+    inertial_coords[i] = logical_coords[i];
+  }
+
+  Variables<tmpl::list<::Tags::dt<Var1>, ::Tags::dt<Var2<Dim>>>>
+      dt_evolved_vars{mesh.number_of_grid_points(), 0.0};
+
+  const TimeStepId time_step_id{true, 3, Time{Slab{0.2, 3.4}, {3, 100}}};
+  const TimeStepId next_time_step_id{true, 3, Time{Slab{0.2, 3.4}, {3, 100}}};
```

Comment:
Surprising that these two TimeStepId are the same even though the names suggest one should come after the other... is that intentional? if so, can you add a comment explaining why?